ant-cli/src/actions/download.rs

```diff
@@ -195,24 +203,14 @@ async fn download_pub_archive_to_disk(
     }
 }
 
-async fn download_public_single_file(
+
+/// Download a single file using already-downloaded data bytes
+/// This avoids re-downloading when we already have the data from a failed archive deserialization
+async fn download_public_single_file_with_data(
```

Comment:
this function is now actually just flush the bytes into disk as a file, no longer `download` anything from network.

better rename to `flush_single_file_data_to_disk` to avoid misleading?